### DISCUSSION TITLE
Actor error

### DISCUSSION THREAD

**Discussion by ArterialMapping** [gis-ops/pyvalhalla#12] [Q&A] [answered]
Any reason why I may be getting this error when running the first actor example? 

![image](https://user-images.githubusercontent.com/85142174/158715509-a7267037-91aa-4fa1-b81d-2d9347e30a42.png)

Example code below.. stuck on "actor = Actor(config)"

from valhalla import Actor, get_config
from valhalla.utils import decode_polyline

import geopandas as gpd
from shapely.geometry import LineString, Polygon, Point
from matplotlib import pyplot as plt
from matplotlib.lines import Line2D
#import contextily as cx

#BASEMAP_SOURCE = cx.providers.CartoDB.VoyagerNoLabels

config = get_config(tile_extract='./custom_files/valhalla_tiles.tar', tile_dir="./custom_files/valhalla_tiles", verbose=False)
actor = Actor(config)

location = actor.locate({"verbose": True, "locations":[{"lat":42.500561,"lon":1.512509}]})
location[0]["edges"][0]

**Comment by nilsnolde** [comment, marked as answer]
I guess you didn’t build any graph before, like the error message is saying. If you read the README, it should clarify things.

**Reply by ArterialMapping** [reply]
Unfortunately, it did not, yesterday I was able to run optimized routes and isochrones on localhost, so I assumed I was good to go to switch over to python?

Do you all have any video tutorials for setting up gis-ops/Valhalla on docker? Guess I set it up wrong.

**Reply by nilsnolde** [reply]
We don’t have videos (yet). If you could route, then the container worked. Then the problem should be somewhere else. 

Can you describe the process you took until you tried the Python bindings, also show the directory tree of custom_files (ie the folder you mapped into the Valhalla container and should hold the graph)

**Reply by nilsnolde** [reply]
I added a line to the README to bring up a valhalla container, build the graph to `./custom_files` and then destroy it. can you give that a try?

after the graph built you should be able to do exactly this:

```
config = get_config(tile_extract='./custom_files/valhalla_tiles.tar'", verbose=True)
actor = Actor(config)
```

**Reply by ArterialMapping** [reply]
So, I don't have an idea as to what the issue was, but I managed to get it working now. 

Now just need to figure out how to do routing with travel times and/or distance and pull that to a shapefile! I did not see anything specific to that in the examples to complete this with CSV rather than JSON, is this something that can be completed with this package?

Actually got it to work with routingpy and geopandas!

**Reply by nilsnolde** [reply]
Great news:) for putting the geometries into a shapefile with python, you’d typically use fiona or some such. Needs a bit of fiddling but you already got the harder part I think

**Comment by nilsnolde** [comment]
Actually we could change the example code so it’s self-contained. If you look into the test directory there’s a tiles.tar of Utrecht, which we could use. Need to clone the repo for that though. I’ll also add a one liner to the documentation to create a graph with docker.
